### PR TITLE
Feature/clean up

### PR DESCRIPTION
**Pull request by NikoBergemann**
### Reason for this PR
- Lagom needs a cleaner API

### Changes in this PR
- ConnectionObjects now are dedicated to single contracts

### CODE REVIEW

File: src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionCourseTrait.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   * Executes the "addCourse" query.
+   *
+   * @param jSonCourse Information about the course to add.
+   * @throws Exception if chaincode throws an exception.
+   * @return Success_state
+   */
+  @throws[TransactionExceptionTrait]
+  @throws[HyperledgerExceptionTrait]
+  def addCourse(jSonCourse: String): String
+  /**
```
Comment (david-buderus) [inline src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionCourseTrait.scala:17]: Use the same amount of new lines between methods and at least one

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala [hunk 2]
```diff
@@ -1,42 +1,39 @@
...
+
+object EnrollmentManager{
+  def enroll(ca_url : String, tlsCert : Path, walletPath : Path, username : String, password : String, organisationId : String): Unit = {
+    // wallet is target for admin certificate
+    val wallet = WalletManager.getWallet(walletPath)
+
+    // check if user already exists in my wallet
+    if (wallet.list.contains(username)) {
+      System.out.println(s"An identity for the admin user $username already exists in the wallet")
+      return
```
Comment (david-buderus) [inline src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: Don't use return in Scala
Use a clear flow

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala [hunk 3]
```diff
@@ -1,42 +1,39 @@
...
+import org.hyperledger.fabric_ca.sdk.{EnrollmentRequest, HFCAClient}
+
+object EnrollmentManager{
+  def enroll(ca_url : String, tlsCert : Path, walletPath : Path, username : String, password : String, organisationId : String): Unit = {
+    // wallet is target for admin certificate
+    val wallet = WalletManager.getWallet(walletPath)
+
+    // check if user already exists in my wallet
+    if (wallet.list.contains(username)) {
+      System.out.println(s"An identity for the admin user $username already exists in the wallet")
```
Comment (david-buderus) [inline src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:17]: `System.out.println` ist Java
Use `println` instead

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala [hunk 4]
```diff
@@ -1,42 +1,39 @@
...
+    caClient.setCryptoSuite(cryptoSuite)
+
+    // enroll my user
+    val enrollmentRequestTLS = new EnrollmentRequest
+    enrollmentRequestTLS.addHost("localhost")
+    enrollmentRequestTLS.setProfile("tls")
+    val enrollment = caClient.enroll(username, password, enrollmentRequestTLS)
+    val identity = Identities.newX509Identity(organisationId, enrollment)
+    wallet.put(username, identity)
+    System.out.println(s"Successfully enrolled user ${username} and imported it into the wallet")
```
Comment (david-buderus) [inline src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:36]: `System.out.println` ist Java
Use `println` instead

File: src/test/scala/CourseAccessTests.scala [hunk 5]
```diff
@@ -1,85 +1,67 @@
...
-import de.upb.cs.uc4.hyperledger.traits.ChaincodeActionsTrait
-import org.scalatest.Succeeded
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AnyWordSpec
+  var chaincodeConnection: ConnectionCourseTrait = _
 
-import scala.util.{Success, Using}
-
-class CourseAccessTests extends AnyWordSpec with Matchers {
+  override def beforeEach() {
```
Comment (david-buderus) [inline src/test/scala/CourseAccessTests.scala:9]: We are using this style:
```scala
override def beforeEach() = {
```

File: src/test/scala/CourseAccessTests.scala [hunk 6]
```diff
@@ -1,85 +1,67 @@
...
+    chaincodeConnection = initializeCourses()
+  }
 
-  val connectionManager = ConnectionManager(
-    Paths.get(getClass.getResource("/connection_profile.yaml").toURI),
-    Paths.get(getClass.getResource("/wallet/").toURI))
+  override def afterEach() {
```
Comment (david-buderus) [inline src/test/scala/CourseAccessTests.scala:13]: We are using this style:
```scala
override def beforeEach() = {
```

File: src/test/scala/CourseErrorTests.scala [hunk 7]
```diff
@@ -1,22 +1,13 @@
...
-
-class CourseErrorTests extends AnyWordSpec with Matchers with BeforeAndAfterEach {
-
-  val connectionManager = ConnectionManager(
-    Paths.get(getClass.getResource("/connection_profile.yaml").toURI),
-    Paths.get(getClass.getResource("/wallet/").toURI))
-  var chaincodeConnection: ChaincodeActionsTrait = null
+  var chaincodeConnection: ConnectionCourseTrait = _
 
   override def beforeEach() {
```
Comment (david-buderus) [inline src/test/scala/CourseErrorTests.scala:9]: We are using this style:
```scala
override def beforeEach() = {
```

File: src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionCourseTrait.scala (hunk 1 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionCourseTrait.scala:17]: nächster push

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala (hunk 3 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:17]: resolved in next push

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala (hunk 4 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:36]: resolved in next push

File: src/test/scala/CourseAccessTests.scala (hunk 5 above)
Comment (NikoBergemann) [reply src/test/scala/CourseAccessTests.scala:9]: jop, hätte m.M.n. sogar der compiler anmeckern sollen

File: src/test/scala/CourseErrorTests.scala (hunk 7 above)
Comment (NikoBergemann) [reply src/test/scala/CourseErrorTests.scala:9]: solved in next push

File: src/test/scala/CourseAccessTests.scala (hunk 6 above)
Comment (NikoBergemann) [reply src/test/scala/CourseAccessTests.scala:13]: solved in next push

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala (hunk 2 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]:
```
if(a){
 handle(a)
 return
}

if(b){
 handle(b)
 return
}

if(c){
handle(c)
return
}

normalFLow()
``` 


find ich schöner als 

```
if(!a){
  if(!b){
   if(!c){
    normalFlow()
   }
   else{
  handle(c)
  }
 else{
  handle(b)
 }
else 
 handle(a)
}
```

Comment (david-buderus) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: If we follow our Code Style, we dont use return.
And in your example case, you maybe could use a match with if statements.

Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: a,b,c can be independent - cases are not always possible:

something like:: 

```
if(a){
 handle(a)
} else {
 if(b){
  handle(b)
 } else {
  if(c){
   handle(c)
  } else {
    normalFLow()
  }
 }
}
```

?

Comment (david-buderus) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: You can use:
```scala
if (a) {
  handle(a)
} else if (b) {
  handle(b)
} else if (c) {
  handle(c)
} else {
  normaleFlow()
}
```

Comment (palkra) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: Can't you just match in the correct order or do I miss something?

Comment (david-buderus) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala:18]: > Can't you just match in the correct order or do I miss something?

In general, yes, you are right.
But in the most cases (at least in Lagom) a, b, c are matchable and than can be checked
with a match which uses if conditions as guards

Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala]: applied in new push

**Pull request merged**